rejson_test.go

```diff
@@ -2,31 +2,229 @@ package rejson
 	if err != nil {
-		t.Fatal("Could not connect to redis.")
+		t.Fatalf("redigo - could not connect to redigo: %v", err)
 		return
 	}
 	defer func() {
 		_, err = conn.Do("FLUSHALL")
+		if err != nil {
+			t.Fatalf("redigo - failed to flush: %v", err)
+		}
 		err = conn.Close()
 		if err != nil {
-			t.Fatal("Failed to communicate to redis-server")
+			t.Fatalf("redigo - failed to close: %v", err)
 		}
 	}()
 
+	t.Run("TestJSONSET", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONSet(rh, t)
+	})
+	t.Run("TestJSONGet", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONGet(rh, t)
+	})
+	t.Run("TestJSONDel", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONDel(rh, t)
+	})
+	t.Run("TestJSONMGet", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONMGet(rh, t)
+	})
+	t.Run("TestJSONType", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONType(rh, t)
+	})
+	t.Run("TestJSONNumIncrBy", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONNumIncrBy(rh, t)
+	})
+	t.Run("TestJSONNumMultBy", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONNumMultBy(rh, t)
+	})
+	t.Run("TestJSONStrAppend", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONStrAppend(rh, t)
+	})
+	t.Run("TestJSONStrLen", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONStrLen(rh, t)
+	})
+	t.Run("TestJSONArrAppend", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONArrAppend(rh, t)
+	})
+	t.Run("TestJSONArrLen", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONArrLen(rh, t)
+	})
+	t.Run("TestJSONArrPop", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONArrPop(rh, t)
+	})
+	t.Run("TestJSONArrIndex", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONArrIndex(rh, t)
+	})
+	t.Run("TestJSONArrTrim", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONArrTrim(rh, t)
+	})
+	t.Run("TestJSONArrInsert", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONArrInsert(rh, t)
+	})
+	t.Run("TestJSONObjLen", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONObjLen(rh, t)
+	})
+	t.Run("TestJSONObjKeys", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONObjKeys(rh, t)
+	})
+	t.Run("TestJSONDebug", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONDebug(rh, t)
+	})
+	t.Run("TestJSONForget", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONForget(rh, t)
+	})
+	t.Run("TestJSONResp", func(t *testing.T) {
+		rh.SetRedigoClient(conn)
+		testJSONResp(rh, t)
+	})
+
+}
+
+func TestGoRedis(t *testing.T) {
+	rh := NewReJSONHandler()
+
+	// GoRedis Test
+	red := goredis.NewClient(&goredis.Options{Addr: "localhost:6379"})
+	defer func() {
+		if err := red.FlushAll().Err(); err != nil {
+			t.Fatalf("goredis - failed to flush: %v", err)
+		}
+		if err := red.Close(); err != nil {
+			t.Fatalf("goredis - failed to communicate to redis-server: %v", err)
+		}
+	}()
+
+	t.Run("TestJSONSet", func(t *testing.T) {
```

Comment:
Wish they had a way to augment the `t.Run()` function with `func(t *testing.T, <extra-args>)`. Could have prevented repeating the test cases for each client.

Reply:
I think the last commit https://github.com/nitishm/go-rejson/pull/29/commits/1c0b597271f6678a81ceb3c086d705e208529292 has fulfilled your wish. 😛